### ISSUE TITLE
SQLite: CREATE TRIGGER unparsable

### ISSUE THREAD

**Issue by skykasko** [sqlfluff/sqlfluff#1846]
## Expected Behaviour

SQLFluff is able to parse `CREATE TRIGGER` statements in SQLite.

## Observed Behaviour

`CREATE TRIGGER` statements raise an error like the following:

<samp>Line 1, Position 1: Found unparsable section: 'CREATE
TRIGGER update_customer_address U...'</samp>

## Steps to Reproduce

Run `sqlfluff lint --dialect sqlite` on a file containing the line (taken from the [SQLite documentation](https://www.sqlite.org/lang_createtrigger.html#some_example_triggers)):
```sql
CREATE TRIGGER update_customer_address UPDATE OF address ON customers
```

## Version
- SQLFluff: 0.8.0
- Python: 3.10.0

**Comment by WittierDinosaur** [comment]
Yeah, looks like the syntax for create trigger is a bit different in SQLite. I think - if you want a quick workaround, if you include one of: BEFORE, AFTER, INSTEAD OF, it should parse

**Comment by skykasko** [comment]
Thanks @WittierDinosaur, but your workaround does not solve the issue for me. If I add `BEFORE` to the example from the SQLite docs like so:

```sql
CREATE TRIGGER update_customer_address BEFORE UPDATE OF address ON customers
  BEGIN
    UPDATE orders SET address = new.address WHERE customer_name = old.name;
  END;
```

Running `sqlfluff lint --dialect sqlite` still results in the error:

<samp>Line 1, Position 1: Found unparsable section: 'CREATE
TRIGGER update_customer_address B...'</samp>

**Comment by WittierDinosaur** [comment]
Ah okay - this is actually going to be a very involved change. Right now we terminate statements on the presence of a(n unquoted) semicolon in the statement. That isn't going to fly here, so we'll need to replicate some of the logic we use in t-sql. Most dialects use `EXECUTE PROCEDURE my_func()` to end their trigger statements - would have been far more convenient if SQLite did the same 😂

**Comment by tunetheweb** [comment]
Made the quick fix to parse the first example.

As @WittierDinosaur says the second (with `BEGIN`/`END`) will take a bit more thought.
